src/components/map/layers/edit/EditNetworkLayer.tsx

```diff
@@ -1,23 +1,76 @@
 import React, { Component } from 'react';
-import { Polyline } from 'react-leaflet';
+import * as L from 'leaflet';
+import { Polyline, withLeaflet } from 'react-leaflet';
+import { matchPath } from 'react-router';
 import { inject, observer } from 'mobx-react';
+import navigator from '~/routing/navigator';
+import SubSites from '~/routing/subSites';
 import { INode, ILink } from '~/models';
-import NodeLocationType from '~/types/NodeLocationType';
 import { EditNetworkStore } from '~/stores/editNetworkStore';
 import TransitTypeColorHelper from '~/util/transitTypeColorHelper';
 import NodeMarker from '../mapIcons/NodeMarker';
+import { LeafletContext } from '../../Map';
 
 interface IEditNetworkLayerProps {
     editNetworkStore?: EditNetworkStore;
+    leaflet: LeafletContext;
 }
 
 @inject('editNetworkStore')
 @observer
 class EditNetworkLayer extends Component<IEditNetworkLayerProps> {
+    editableLinks: L.Polyline[] = [];
+
+    private drawEditableLinks() {
+        const links = this.props.editNetworkStore!.links;
+        const map = this.props.leaflet.map;
+        const isLinkView = Boolean(matchPath(navigator.getPathName(), SubSites.link));
```

Comment:
We currently decide isLinkView in multiple places in the file, maybe we could set a private class variable with this data instead. Which could be set in componentDidUpdate && componentDidMount?

Reply:
isLinkViewVisible now being used once at render